pyccel/parser/semantic.py

```diff
@@ -3785,209 +3875,249 @@
+                                                       [Assign(var, a.var)])))
+            body.insert2body(*optional_inits, back=False)
+
+            func_kwargs = {
+                    'global_vars':global_vars,
+                    'is_pure':is_pure,
+                    'is_elemental':is_elemental,
+                    'is_private':is_private,
+                    'imports':imports,
+                    'decorators':decorators,
+                    'is_recursive':is_recursive,
+                    'functions': sub_funcs,
+                    'interfaces': func_interfaces,
+                    'result_pointer_map': result_pointer_map,
+                    'docstring': docstring,
+                    'scope': scope,
+                     'is_annotated':True,
+                    'syntactic_node':expr,
+            }
+            if is_inline:
+                func_kwargs['namespace_imports'] = namespace_imports
+                global_funcs = [f for f in body.get_attribute_nodes(FunctionDef) if self.scope.find(f.name, 'functions')]
+                func_kwargs['global_funcs'] = global_funcs
+                cls = InlineFunctionDef
+            else:
+                cls = FunctionDef
+            func = cls(name,
+                    arguments,
+                    results,
+                    body,
+                    **func_kwargs)
+            if not is_recursive:
+                recursive_func_obj.invalidate_node()
 
-                # Find all nodes which can modify variables
-                assigns = body.get_attribute_nodes(Assign, excluded_nodes = (FunctionCall,))
-                calls   = body.get_attribute_nodes(FunctionCall)
+            if cls_name:
+                # update the class methods
+                if not is_interface:
+                    bound_class.add_new_method(func)
 
-                # Collect the modified objects
-                lhs_assigns   = [a.lhs for a in assigns]
-                modified_args = [call_arg.value for f in calls
-                                    for call_arg, func_arg in zip(f.args, f.funcdef.arguments) if func_arg.inout]
-                # Collect modified variables
-                all_assigned = [v for a in (lhs_assigns + modified_args) for v in
-                                (a.get_attribute_nodes(Variable) if not isinstance(a, Variable) else [a])]
+            funcs += [func]
 
-                # ... computing inout arguments
-                for a in arguments:
-                    if a.name not in chain(results_names, ['self']) and a.var not in all_assigned:
-                        a.make_const()
-                # ...
+        if function_call is not None and len(funcs) == 0:
+            for args in annotated_args:
+                #raise errors if we do not find any compatible function def
+                self._check_argument_compatibility(function_call, args, expr, is_elemental, error=True)
+            return EmptyNode()
```

Comment:
\bot accept , it is tested in [ea7825a](https://github.com/pyccel/pyccel/pull/1728/commits/ea7825a04767dcc0d416f277c74478f581302fbd)

Reply:
You don't need to use the `bot accept` command if you add tests later. When you rerun the bot it will ask you to resolve the conversation for all the places where the code has been run. In this case it didn't notice this. Are you sure that [ea7825a](https://github.com/pyccel/pyccel/pull/1728/commits/ea7825a04767dcc0d416f277c74478f581302fbd) adds a test for this? Perhaps the code doesn't quite work like you think it does? Or an error is raised before the empty node is reached?

Reply:
yes the error is raised before the EmptyNode, but then that is outside the for loop, that part is reached if we do not have an error